recipes/tcl/8.6.9/conanfile.py

```diff
@@ -0,0 +1,220 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanExceptionInUserConanfileMethod
+from conans.util.env_reader import get_env
+import os
+import shutil
+import tempfile
+
+
+class TclConan(ConanFile):
+    name = "tcl"
+    description = "Tcl is a very powerful but easy to learn dynamic programming language."
+    topics = ("conan", "tcl", "scripting", "programming")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tcl.tk"
+    license = "TCL"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "shared": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": False,
+    }
+    _source_subfolder = "sources"
+    requires = ("zlib/1.2.11")
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _install_folder(self):
+        # conan-center-index forbids having share/man folders at the package root
+        # it also advises to put everything into a bin folder
+        # until there is a better way, let's go for the bin/bin weirdness
+        return os.path.join(self.package_folder, "bin").replace("\\", "/")
+
+    def configure(self):
+        if self.settings.compiler != "Visual Studio":
```

Comment:
why do you need to remove only for MSVC ?

Reply:
I have no idea, I took the recipe from bincrafters' since it was way more complete than mine. I don't know who the original author is though

Reply:
I wrote it originally. I don't remember the reason either. It probably is because the `Visual Studio` build was an afterthought.
The only c++ sources are in the `compat/zlib` subdirectory.
So I guess it would be safe to remove them unconditionally.

A test to see whether the conan zlib is used would be to remove a zlib core source file (e.g. `compat/zlib/inflate.c`) and see whether the build succeeds.

Reply:
@madebr I really don't feel confident fiddling with both Autotools/NMAKE to make it work without the `compat/zlib` folder. Nmake already drove me crazy today 🙃 .

Reply:
I share the pain. It took me also a while before I got it working. And barely, so it appears now.

Reply:
Haha, and wait for the next recipe: `sqlcipher`...